File: trunk/libwarc/warc-tools/lib/private/plugin/ruby/wrecord.rb
# -------------------------------------------------------------------  #
# Copyright (c) 2007-2008 Hanzo Archives Limited.                      #
#                                                                      #
# Licensed under the Apache License, Version 2.0 (the "License");      #
# you may not use this file except in compliance with the License.     #
# You may obtain a copy of the License at                              #
#                                                                      #
#     http//www.apache.org/licenses/LICENSE-2.0                        #
#                                                                      #
# Unless required by applicable law or agreed to in writing, software  #
# distributed under the License is distributed on an "AS IS" BASIS,    #
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or      #
# implied.                                                             #
# See the License for the specific language governing permissions and  #
# limitations under the License.                                       #
#                                                                      #
# You may find more information about Hanzo Archives at                #
#                                                                      #
#     http//www.hanzoarchives.com/                                     #
#                                                                      #
# You may find more information about the WARC Tools project at        #
#                                                                      #
#     http//code.google.com/p/warc-tools/                              #
# -------------------------------------------------------------------  #

require 'warctools'


class WRecord

  @me = 0
  @type = :None

##Constructor ##
  def initialize
      @me = Warctools.bless_WRecord()
     if (@me == nil) then
        raise RuntimeError, "Can not create a WRecord object."
     end
      @type = :WRecord
  end


##The getters ##

  def getWarcId()
     return Warctools.WRecord_getWarcId(@me)
  end


  def getRecordType()
     return Warctools.WRecord_getRecordType(@me)
  end


  def getTargetUri()
     return Warctools.WRecord_getTargetUri(@me)
  end


  def getDate()
     return Warctools.WRecord_getDate(@me)
  end


  def getContentType()
     return Warctools.WRecord_getContentType(@me)
  end


  def getRecordId()
     return Warctools.WRecord_getRecordId(@me)
  end


  def getContent()
     return Warctools.WRecord_getContent(@me)
  end


  def getAnvlValue (key)
     return Warctools.WRecord_getAnvlValue(@me,key)
  end


  def getContentLength()
     return Warctools.WRecord_getContentLength(@me)
  end


  def getConcurrentTo()
     return Warctools.WRecord_getConcurrentTo(@me)
  end


  def getBlockDigest()
     return Warctools.WRecord_getBlockDigest(@me)
  end


  def getPayloadDigest()
     return Warctools.WRecord_getPayloadDigest(@me)
  end


  def getIpAddress()
     return Warctools.WRecord_getIpAddress(@me)
  end


  def getRefersTo()
     return Warctools.WRecord_getRefersTo(@me)
  end


   def getTruncated()
     return Warctools.WRecord_getTruncated(@me)
  end


  def getWarcInfoId()
     return Warctools.WRecord_getWarcInfoId(@me)
  end


  def getProfile()
     return Warctools.WRecord_getProfile(@me)
  end


  def getPayloadType()
     return Warctools.WRecord_getPayloadType(@me)
  end


  def getSegmentOriginId()
     return Warctools.WRecord_getSegmentOriginId(@me)
  end


  def getSegmentNumber()
     return Warctools.WRecord_getSegmentNumber(@me)
  end


   def getSegTotalLength()
     return Warctools.WRecord_getSegTotalLength(@me)
  end


  def getFileName()
     return Warctools.WRecord_getFilename(@me)
  end


  def getOffset()
     return Warctools.WRecord_getOffset(@me)
  end


  def getUncompressedSize()
     return Warctools.WRecord_getUncompressedSize(@me)
  end


  def getCompressedSize()
     return Warctools.WRecord_getCompressedSize(@me)
  end



  def getBloc(wfile, with_http, code)
      if (wfile.type == :WFile) then
         withh = Warctools::WARC_TRUE

         if (withh_http == false) then
            withh = Warctools::WARC_FALSE
         end

         return Warctools.WRecord_getBloc(@me, wfile.getInternal(), withh, code)
      else
         return ""
     end
  end


  def getAnvlFieldsNumber()
     return (Warctools.WRecord_getAnvlFieldsNumber(@me))
  end


  def getAnvlFieldKey(rank)
      key = Warctools.getFieldKey(@me, rank)

     if (key == nil) then
         return ""
     else
        return (key)
     end
  end


  def getAnvlFieldValue(rank)
     val = Warctools.getFieldValue(@me, rank)

     if (val == nil)
        return ""
     else
        return (val)
     end
  end



##The setters ##

  def setRecordType(type)
     return Warctools.WRecord_setRecordType(@me,type)
  end


  def setTargetUri(uri, uri_len)
     return Warctools.WRecord_setTargetUri(@me, uri, uri_len)
  end

  def setDate(date, date_len)
     return Warctools.WRecord_setDate(@me, date, date_len)
  end


  def setContentType(ctype, ctype_len)
     return Warctools.WRecord_setContentType(@me, ctype, ctype_len)
  end


  def setRecordId(rid, id_len)
     return Warctools.WRecord_setRecordId(@me, rid, id_len)
  end


  def setConcurrentTo(rid, id_len)
     return Warctools.WRecord_setConcurrentTo(@me, rid, id_len)
  end


  def setBlockDigest(digest, digest_len)
     return Warctools.WRecord_setBlockDigest(@me, digest, digest_len)
  end


  def setPayloadDigest(digest, digest_len)
     return Warctools.WRecord_setPayloadDigest(@me, digest, digest_len)
  end


  def setIpAddress(ip, ip_len)
     return Warctools.WRecord_setIpAddress(@me, ip, ip_len)
  end


  def setRefersTo(rid, id_len)
     return Warctools.WRecord_setRefersTo(@me, rid, id_len)
  end


  def setTruncated(cause, cause_len)
     return warc.WRecord_setTruncated(@me, cause, cause_len)
  end


  def setWarcInfoId(rid, id_len)
     return Warctools.WRecord_setWarcInfoId(@me, rid, id_len)
  end


  def setProfile(profile, profile_len)
     return Warctools.WRecord_setProfile(@me, profile, profile_len)
  end


  def setPayloadType(ptype, type_len)
     return Warctools.WRecord_setPauloadType(@me, ptype, type_len)
  end


  def setSegmentOriginId(rid, id_len)
     return Warctools.WRecord_setSegmentOriginId(@me, rid, id_len)
  end


  def setSegmentNumber(num)
     return warc.WRecord_setSegmentNumber(@me, num)
  end


  def setSegTotalLength(slen)
     return Warctools.WRecord_setSegTotalLength(@me, slen)
  end


  def setFilename(name, name_len)
     return Warctools.WRecord_setFilename(@me, name, name_len)
  end


  def addAnvl(key, key_len, value, value_len)
     return Warctools.WRecord_addAnvl(@me, key, key_len, value, value_len)
  end


  def setContentFromFileName(name)
     return Warctools.WRecord_setContentFromFileName(@me, name)
  end


  def setDateFromArc(date, date_len)
     return Warctools.WRecord_setDateFromArc(@me, date, date_len)
  end

  def setContentFromString(str_v, str_len)
     return Warctools.WRecord_setContentFromString(@me, str_v, str_len)
  end


  def setContentFromStringConcat(str_v, str_len)
     return Warctools.WRecord_setContentFromStringConcat(@me, str_v, str_len)
  end


##Ruby WRecord class methods ##

  def external_set(w,r)
     if (w.type == :WFile) then
         @me = r 
     end
  end


  def getInternal (w)
     if (w.type == :WFile || w.type == :WBloc || w.type == :ARecord) then
        return @me
     else
       return (nil)
     end
  end
 

  def type
     return @type
  end

## destructor ##

  def destroy()
     Warctools.destroy(@me)
  end


##  end of wrapper ##

end

File: trunk/libwarc/warc-tools/lib/private/plugin/python/wrecord.py
# -------------------------------------------------------------------  #
# Copyright (c) 2007-2008 Hanzo Archives Limited.                      #
#                                                                      #
# Licensed under the Apache License, Version 2.0 (the "License");      #
# you may not use this file except in compliance with the License.     #
# You may obtain a copy of the License at                              #
#                                                                      #
#     http://www.apache.org/licenses/LICENSE-2.0                       #
#                                                                      #
# Unless required by applicable law or agreed to in writing, software  #
# distributed under the License is distributed on an "AS IS" BASIS,    #
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or      #
# implied.                                                             #
# See the License for the specific language governing permissions and  #
# limitations under the License.                                       #
#                                                                      #
# You may find more information about Hanzo Archives at                #
#                                                                      #
#     http://www.hanzoarchives.com/                                    #
#                                                                      #
# You may find more information about the WARC Tools project at        #
#                                                                      #
#     http://code.google.com/p/warc-tools/                             #
# -------------------------------------------------------------------  #

import warc
import types
import sys


sys.path.insert (0, ".")
from wtypes import WTypes

#class WTypes:
	#"A class to define Python WARC classes types"
	#def __init__(self):
		#self.WNone   = 0
		#self.WFile   = 1
		#self.WRecord = 2
		#self.WBloc   = 3
		#self.AFile   = 4
		#self.ARecord = 5
		

class WRecord:

##Constructor ##
	def __init__(self):
		self.classtype = WTypes()
		self.me = warc.bless_WRecord()
		
		self.type = self.classtype.WRecord


##The getters ##

	def getWarcId(self):
		return warc.WRecord_getWarcId(self.me)


	def getRecordType(self):
		return warc.WRecord_getRecordType(self.me)

	########
	def getTargetUri(self):
		return warc.WRecord_getTargetUri(self.me)


	def getDate(self):
		return warc.WRecord_getDate(self.me)

	########
	def getContentType(self):
		return warc.WRecord_getContentType(self.me)


	def getRecordId(self):
		return warc.WRecord_getRecordId(self.me)


	def getContent(self):
		return warc.WRecord_getContent(self.me)


	def getAnvlValue (self, key):
		return warc.WRecord_getAnvlValue(self.me,key)


	def getContentLength(self):
		return warc.WRecord_getContentLength(self.me)

	########
	def getConcurrentTo(self):
		return warc.WRecord_getConcurrentTo(self.me)

	########
	def getBlockDigest(self):
		return warc.WRecord_getBlockDigest(self.me)

	########
	def getPayloadDigest(self):
		return warc.WRecord_getPayloadDigest(self.me)

        ########
	def getIpAddress(self):
		return warc.WRecord_getIpAddress(self.me)

        ########
	def getRefersTo(self):
		return warc.WRecord_getRefersTo(self.me)

	########
	def getTruncated(self):
		return warc.WRecord_getTruncated(self.me)

 	########
	def getWarcInfoId(self):
		return warc.WRecord_getWarcInfoId(self.me)

	########
	def getProfile(self):
		return warc.WRecord_getProfile(self.me)

	########
	def getPayloadType(self):
		return warc.WRecord_getPayloadType(self.me)

	########
	def getSegmentOriginId(self):
		return warc.WRecord_getSegmentOriginId(self.me)

	########
	def getSegmentNumber(self):
		return warc.WRecord_getSegmentNumber(self.me)

	########
 	def getSegTotalLength(self):
		return warc.WRecord_getSegTotalLength(self.me)

	########
	def getFileName(self):
		return warc.WRecord_getFilename(self.me)


	def getOffset(self):
		return warc.WRecord_getOffset(self.me)


	def getUncompressedSize(self):
		return warc.WRecord_getUncompressedSize(self.me)


	def getCompressedSize(self):
		return warc.WRecord_getCompressedSize(self.me)


	def getBloc(self, with_http, wfile, code):
		if (wfile.type() == classtype.WFile):
			withh = warc.cvar.WARC_TRUE
			if (withh_http == false):
				withh = warc.cavar.WARC_FALSE 
			
			return warc.WRecord_getBloc(self.me, wfile.getInternal(self), withh, code)
		else:
			return ""


	def getAnvlFieldsNumber(self):
		return (warc.WRecord_getAnvlFieldsNumber(self.me))


	def getAnvlFieldKey(self, rank):
		key = warc.getFieldKey (self.me, rank)
		if (not (key)):
			return ""
		else:
			return (key)


	def getAnvlFieldValue(self, rank):
		val = warc.getFieldValue (self.me, rank)
		if (not (val)):
			return ""
		else:
			return (val)



##The setters ##

	def setRecordType(self,type):
		return warc.WRecord_setRecordType(self.me,type)


	def setTargetUri(self, uri, uri_len):
		return warc.WRecord_setTargetUri(self.me, uri, uri_len)


	def setDate(self, date, date_len):
		return warc.WRecord_setDate(self.me, date, date_len)


	def setContentType(self, ctype, ctype_len):
		return warc.WRecord_setContentType(self.me, ctype, ctype_len)


	def setRecordId(self, rid, id_len):
		return warc.WRecord_setRecordId(self.me, rid, id_len)


	def setConcurrentTo(self, rid, id_len):
		return warc.WRecord_setConcurrentTo(self.me, rid, id_len)


	def setBlockDigest(self, digest, digest_len):
		return warc.WRecord_setBlockDigest(self.me, digest, digest_len)


	def setPayloadDigest(self, digest, digest_len):
		return warc.WRecord_setPayloadDigest(self.me, digest, digest_len)


	def setIpAddress(self, ip, ip_len):
		return warc.WRecord_setIpAddress(self.me, ip, ip_len)


	def setRefersTo(self, rid, id_len):
		return warc.WRecord_setRefersTo(self.me, rid, id_len)


	def setTruncated(self, cause, cause_len):
		return warc.WRecord_setTruncated(self.me, cause, cause_len)


	def setWarcInfoId(self, rid, id_len):
		return warc.WRecord_setWarcInfoId(self.me, rid, id_len)


	def setProfile(self, profile, profile_len):
		return warc.WRecord_setProfile(self.me, profile, profile_len)


	def setPayloadType(self, ptype, type_len):
		return warc.WRecord_setPayloadType(self.me, ptype, type_len)


	def setSegmentOriginId(self, rid, id_len):
		return warc.WRecord_setSegmentOriginId(self.me, rid, id_len)


	def setSegmentNumber(self, num):
		return warc.WRecord_setSegmentNumber(self.me, num)


	def setSegTotalLength(self, slen):
		return warc.WRecord_setSegTotalLength(self.me, slen)


	def setFilename(self, name, name_len):
		return warc.WRecord_setFilename(self.me, name, name_len)


	def addAnvl(self, key, key_len, value, value_len):
		return warc.WRecord_addAnvl(self.me, key, key_len, value, value_len)


	def setContentFromFileName(self, name):
		return warc.WRecord_setContentFromFileName(self.me, name)


	def setDateFromArc(self, date, date_len):
		return warc.WRecord_setDateFromArc(self.me, date, date_len)
	
	def setContentFromString(self, str_v, str_len):
		return warc.WRecord_setContentFromString (self.me, str_v, str_len)
	
	
	def setContentFromStringConcat(self, str_v, str_len):
		return warc.WRecord_setContentFromStringConcat (self.me, str_v, str_len)


##Python WRecord class methods ##

	def external_set(self, w,r):
		if (w.type == self.classtype.WFile):
			self.me = r 

	def getInternal (self, w):
		if (w.type == self.classtype.WFile or w.type == self.classtype.WBloc or w.type == self.classtype.ARecord):
			return self.me
		else:
			return 0

 

	def type(self):
		return self.type
 

## destructor ##
	def destroy(self):
		warc.destroy(self.me)


##  end of wrapper ##
